diff -ur librsync-1.0.0/Makefile.am librsync-1.0.0-new/Makefile.am
--- librsync-1.0.0/Makefile.am	2015-01-23 16:39:09.000000000 +0000
+++ librsync-1.0.0-new/Makefile.am	2015-10-17 11:01:47.248900223 +0000
@@ -26,9 +26,9 @@
 	depcomp install-sh ltmain.sh missing mkinstalldirs \
 	prototab.c prototab.h
 
-EXTRA_DIST = autogen.sh configure.msc $(noinst_SCRIPTS) rdiff.magic \
+EXTRA_DIST = autogen.sh configure.msc $(noinst_SCRIPTS) \
 	librsync.spec README.RPM libversions.txt \
-	PCbuild/rdiff.dsp PCbuild/PCbuild.dsw   PCbuild/config.h \
+	PCbuild/PCbuild.dsw   PCbuild/config.h \
 	PCbuild/librsync-config.h
 
 include_HEADERS = librsync.h librsync-config.h
@@ -53,16 +53,6 @@
 # This is the default for any programs that don't specify a preference.
 LDADD = librsync.la
 
-# This is the rdiff application
-bin_PROGRAMS = rdiff
-
-rdiff_SOURCES = rdiff.c isprefix.c isprefix.h librsync.h librsync-config.h \
-	trace.h
-
-rdiff_LDADD = librsync.la
-
-rdiff_DEPENDENCIES = librsync.la
-
 # this script is used to build prototab.[ch]
 noinst_SCRIPTS = mkprototab.pl
 
